tests/envs/test_make.py

```diff
@@ -146,7 +140,6 @@ def test_apply_step_compatibility():
         max_episode_steps=3,
     )
     env = gym.make("testing-old-env")
-    assert has_wrapper(env, StepAPICompatibility)
```

Comment:
Why is this assert removed?

Reply:
The old semantic for the compatibility layer was that it's a valid `gym.Wrapper`, because a `gym.Env` could be both in the old and new API. Now for an env to be a valid `gym.Env`, it has to follow the new API, which means that something written with the old API, isn't technically a `gym.Env`, i.e. it can't be the input to a `gym.Wrapper`.

The new compatibility layer is simply an implementation of a `gym.Env`, which happens to have a somewhat similar structure to a wrapper, but isn't one. So it used to be the case that an old-to-new environment is wrapped by StepAPICompatibility as a wrapper, which was checked in the test. Now that's not the case anymore.

Reply:
Should the environment not come before the passive environment checker?